### ISSUE TITLE
Signed in with valid FxAccount but details do not appear on Account Settings view

### ISSUE THREAD

**Issue by isabelrios** [mozilla-lockbox/lockbox-ios#392]
iOS 10.3.3 and 11.3.1
Lockbox: 1139

Steps:
1- Log in with a valid account
2- Go to Settings -> Account

Actual:
Under the image place holder it shows Firefox Account instead of my account details
Going back to Firefox Lockbox screen, saved logins do not appear there. Trying to refresh is possible to see like grey lines where logins should appear for a while.
Please see video here: https://youtu.be/-bCQ8jzeH-g
![account settings](https://user-images.githubusercontent.com/1897507/40224595-ef16b40c-5a86-11e8-9167-847135d1223f.PNG)

Expected:
FxAccount's details are shown. ~Saved logins should appear.~ (because of confirmation issue)

I'm happy to help sharing logs or what is needed. Will try to get more information.

**Comment by devinreams** [comment]
Repurposed to be simply about the FxAccount details. 

Will track the problem with not knowing a confirmation is needed at: #328

**Comment by devinreams** [comment]
Via @jhugman: feels like something in DataStore or fxa-sync library. Finding whatever listener or message or whatever isn't happening will likely fix and improve a lot...

**Comment by jhugman** [comment]
Both @sashei & I have seen this working in Berlin, however, I'm concerned that this is to do with the FxAiOSClientID we changed shortly after returning from Berlin. 

Have sent an email to FxA.

**Comment by devinreams** [comment]
>I'm concerned that this is to do with the FxAiOSClientID we changed shortly after returning from Berlin.

@jhugman Good thought!

I just tried changing the `ClientID` back at https://github.com/mozilla-lockbox/lockbox-ios-fxa-sync/commit/523e910d217229c159263c375ba7a8d231ba40da (and updated my local Cartfile/resolved files) and I'm now seeing FxA profile details on the settings screen. 😆 

https://github.com/mozilla-lockbox/lockbox-ios-fxa-sync/commit/523e910d217229c159263c375ba7a8d231ba40da

I'll submit a PR as a _backup_, just in case the magical switch the FxA team needs to flip (may or may not exist) can/does not get flipped.

**Comment by devinreams** [comment]
OK I've seen this work (but can't get it working now, unsure why...).

@sashei has also seen the profile details load on a test device against `master`

@nickbrandt @changecourse @sandysage can you all please try downloading a recent build from `master`, and check if the profile details show up in the account settings screen for any of you? (and try disconnecting and signing in again if not, and then try deleting the app and then re-installing again if not)

**Comment by isabelrios** [comment]
I've seen the test account name there. But looks like it is not consistent, I mean it appeared, I uninstall the app/disconnected the account and wasn't shown and doing the same was shown again...but it was in a branch two or three commits behind master (the xcuitest branch) Will check on latest master too...

**Comment by devinreams** [comment]
I tried a bunch more locally with Simulator and by using different accounts this got "unstuck" and is working for me.

I tried deleting my app on my device and re-installing and it still won't show actual profile details for me, for some reason...

I'm curious to get more data points from other testers.

Either way, it feels like the server side stuff is there/enabled now 👍 but the app/client side is handling it inconsistently 👎 perhaps?

We do have a follow-up story to refresh FxA info that may be useful to re-consider sooner than later: #198

**Comment by sandysage** [comment]
@devinreams I also tried with build 1383 and no profile info. I reinstalled the app. Created a new profile. Disconnected and reconnected. 👎

**Comment by isabelrios** [comment]
On latest master: 6058922
<img width="284" alt="captura de pantalla 2018-05-31 a las 22 47 35" src="https://user-images.githubusercontent.com/1897507/40807798-9f9a5cf4-6525-11e8-94dd-08b8898928b7.png">

**Comment by jhugman** [comment]
This looks like it needs https://bugzilla.mozilla.org/show_bug.cgi?id=1460646 to be resolved.

Once the work is done there, this should fix itself with no further action on the iOS side.

**Comment by devinreams** [comment]
>this should fix itself with no further action on the iOS side.

The server side change to accept both `mozilla-lockbox.github.io` and `lockbox.firefox.com` redirects is complete. The app flags are also changed.

While I do see profile data load in to the settings screen fairly regularly, it's still inconsistent on my personal iPhone X (but others have reported it working just fine on their phones). I _think_ this may be a result of some aggressive asset caching (I once saw the old "Lockbox", sans "Firefox" Loading screen asset) so will chalk it up to that.

I think we can close this (please let me know if you disagree) and if we get many reports this isn't loading for others we should re-open to troubleshoot or consider prioritizing #198

**Comment by nickbrandt** [comment]
So this still appears to be a current issue with the latest build. New users downloading the app from Test Pilot (Blake Winton for example), are not seeing their account details on the account screen. Both @changecourse and I are also still experiencing the issue (initially this was thought to be a caching issue).

**Comment by devinreams** [comment]
>this still appears to be a current issue with the latest Test Flight build. New users downloading the app from Test Flight (Blake Winton for example), are not seeing their account details on the account screen

@jhugman do you feel this is still on the FxA side? something to do with messaging for the profile details not going through? because I do see it on Simulator (so code seems fine) but not on device. 😔 

What do you suggest we can do to troubleshoot a bit more?

**Comment by devinreams** [comment]
The server-side fix should also remedy this once and for all:
https://github.com/mozilla/fxa-auth-server/issues/2491
